src/ngraph/op/fused/clamp.cpp

```diff
@@ -18,12 +18,110 @@
 #include "ngraph/builder/make_constant.hpp"
 #include "ngraph/op/maximum.hpp"
 #include "ngraph/op/minimum.hpp"
+#include "ngraph/runtime/reference/clamp.hpp"
+#include "ngraph/util.hpp"
 
 using namespace std;
 using namespace ngraph;
 
 constexpr NodeTypeInfo op::Clamp::type_info;
 
+namespace
+{
+    template <element::Type_t ET, typename T>
+    bool evaluate(const HostTensorPtr& arg, const HostTensorPtr& out, T min, T max, size_t count)
+    {
+        runtime::reference::clamp<T>(
+            arg->get_data_ptr<ET>(), out->get_data_ptr<ET>(), min, max, count);
+        return true;
+    }
+
+    bool evaluate_clamp(
+        const HostTensorPtr& arg, const HostTensorPtr& out, double min, double max, size_t count)
+    {
+        auto ceil_func = [](double x) { return ceil(x); };
+        auto floor_func = [](double x) { return floor(x); };
+
+        bool rc = true;
+        switch (arg->get_element_type())
+        {
+            TYPE_CASE(i8)
+            (arg,
+             out,
+             double_to_int<int8_t>(min, ceil_func),
+             double_to_int<int8_t>(max, floor_func),
+             count);
+            break;
+            TYPE_CASE(i16)
+            (arg,
+             out,
+             double_to_int<int16_t>(min, ceil_func),
+             double_to_int<int16_t>(max, floor_func),
+             count);
+            break;
+            TYPE_CASE(i32)
+            (arg,
+             out,
+             double_to_int<int32_t>(min, ceil_func),
+             double_to_int<int32_t>(max, floor_func),
+             count);
+            break;
+            TYPE_CASE(i64)
+            (arg,
+             out,
+             double_to_int<int64_t>(min, ceil_func),
+             double_to_int<int64_t>(max, floor_func),
+             count);
+            break;
+            TYPE_CASE(u8)
+            (arg,
+             out,
+             double_to_int<uint8_t>(min, ceil_func),
+             double_to_int<uint8_t>(max, floor_func),
+             count);
+            break;
+            TYPE_CASE(u16)
+            (arg,
+             out,
+             double_to_int<uint16_t>(min, ceil_func),
+             double_to_int<uint16_t>(max, floor_func),
+             count);
+            break;
+            TYPE_CASE(u32)
+            (arg,
+             out,
+             double_to_int<uint32_t>(min, ceil_func),
+             double_to_int<uint32_t>(max, floor_func),
+             count);
+            break;
+            TYPE_CASE(u64)
+            (arg,
+             out,
+             double_to_int<uint64_t>(min, ceil_func),
+             double_to_int<uint64_t>(max, floor_func),
+             count);
+            break;
+            TYPE_CASE(f16)(arg, out, static_cast<float16>(min), static_cast<float16>(max), count);
```

Comment:
Here the casts of large values should go to the fp16/bfloat16 infinities, but you need to test that it works.

Reply:
Done.